identity/aziot-cloud-client-async-common/src/lib.rs

```diff
@@ -30,18 +31,15 @@ pub async fn get_sas_connector(
         let sig_data = format!("{}\n{}", &resource_uri, expiry);
 
         let signature = key_client
-            .sign(
-                &key_handle,
-                aziot_key_common::SignMechanism::HmacSha256,
-                sig_data.as_bytes(),
-            )
+            .sign_with_key(&key_handle, sig_data.as_bytes())
             .await?;
 
         let signature = base64::encode(&signature);
 
         let token = url::form_urlencoded::Serializer::new(format!("sr={}", resource_uri))
             .append_pair("sig", &signature)
             .append_pair("se", &expiry)
+            .append_pair("skn", "registration")
```

Comment:
This isn't required for non-TPM DPS registration or IoT Hub connection. This should only be set when provisioning type is Dps and attestation type is Tpm, right?

Reply:
The old repo unconditionally set it (see https://github.com/Azure/iotedge/blob/7831fc5ad76fa3bd33b1dc1e4630ac4db771e9ff/edgelet/dps/src/registration.rs#L84), so it seems fine leaving it in?

Reply:
Sure, but it applies to all DPS requests. Currently, you have it for all tokens (including hub).

Reply:
Yeah, you're right. I'm trying to think of the best way to fix this... The quick-and-dirty approach would be to add some sort of `is_tpm_registration: bool` to the method, but that's pretty ugly. Any suggestions?

Reply:
Yes - that is what I was thinking of too. Please change it because it might cause inadvertent failures in IoT Hub client logic if `skn` becomes a valid parameter.

Reply:
I added the boolean, and checked that provisioning still works :+1: